From ea4609a982f173cfddb789e5819cf14006bbf7b4 Mon Sep 17 00:00:00 2001
From: "Richard A. Griffiths" <richard.a.griffiths@intel.com>
Date: Thu, 13 Aug 2015 12:34:10 -0700
Subject: [PATCH 17/28] input: silead: correct id mask for multitouch

Multitouch was not being detected because the id was always 0.
Removed the unnecessary bit shift.

Change-Id: I5b0b3f77154b217a1e3261a27321225e8a927463
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-25968
Signed-off-by: Richard A. Griffiths <richard.a.griffiths@intel.com>
Reviewed-on: https://android.intel.com:443/472780
Reviewed-on: https://android.intel.com:443/483147
---
 drivers/input/touchscreen/silead.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/silead.c b/drivers/input/touchscreen/silead.c
index eb36841..c7a4255 100644
--- a/drivers/input/touchscreen/silead.c
+++ b/drivers/input/touchscreen/silead.c
@@ -189,7 +189,7 @@ static void silead_ts_read_data(struct i2c_client *client)
 
 		/* Bits 4-7 are the touch id */
 		id = (buf[offset + SILEAD_POINT_X_MSB_OFF] &
-		      SILEAD_TOUCH_ID_MASK) >> 4;
+		      SILEAD_TOUCH_ID_MASK);
 
 		/* Bits 0-3 are MSB of X */
 		buf[offset + SILEAD_POINT_X_MSB_OFF] = buf[offset +
-- 
1.9.1

